builder/optimizations.cc

```diff
@@ -280,7 +280,7 @@ class buffer_aliaser : public node_mutator {
         }
       }
       if (op->dims[d].stride.defined()) {
-        if (!prove_true(op->dims[d].stride == alias_dim.stride)) {
+        if (!prove_true(op->dims[d].stride == alias_dim.stride) && !prove_true(op->dims[d].stride == target_info.dims[alias.permutation[d]].stride)) {
```

Comment:
Generally I agree it would be better to do this in the ```alias_buffers```, so #464 might be a better solution. On the other hand, while working on this I started wondering if it'd make to do some of the aliasing early and move parts of the ```alias_buffer``` to before IR construction.